server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -71,10 +73,14 @@ private void handleUpdate(String mailboxId, MailboxUpdateRequest updateRequest,
         try {
             validateMailboxName(updateRequest, mailboxSession);
             Mailbox mailbox = getMailbox(mailboxId, mailboxSession);
+            checkRole(mailbox.getRole());
```

Comment:
I don't like this code snipset much...

I would prefer a plain old else / if....

IMO, if you throw an exception, it means your class encountered a problem it can not solve, and relies on the caller to handle it. Here you throw an exception in a method, that is handled by an other pethod, which means your class knows how to handle the exception.

http://stackoverflow.com/questions/8621762/java-if-vs-try-catch-overhead

The important part for me is :

```
try/catch is for things that go wrong that are outside of your control and not in the normal program flow. 


if statements should be normal flow and ordinary error checking. So, for example, user fails to populate
 a required input field? Use if for that, not try/catch.
```